### DISCUSSION TITLE
Martensite Parent Grain Reconstruction

### DISCUSSION THREAD

**Discussion by HamidKamali** [mtex-toolbox/mtex#817] [Ask Anything] [answered]
Dear All,

I am working on reconstruction of deformation-induced martensite (alpha prime-bcc) into parent austenite. Three phases exist in the microstructure including austenite, alpha prime (bcc) and epsilon (hcp) martensite.
Using the code updated in the Mtex 5.6.0, the reconstruction was conducted. 
The reconstructed map is shown below, where the area of interest is enclosed by red line. 

![Mtex discussion](https://user-images.githubusercontent.com/79619598/109101642-36dbf680-777b-11eb-8851-686facd8c2cb.jpg)

For clarity, the initial IPF map of alpha prime martensite is also shown.

![default1](https://user-images.githubusercontent.com/79619598/109101819-9b975100-777b-11eb-81ef-102404f9a884.jpg)

I believe the area, enclosed by red line, belongs to the same parent austenite grain, however, after reconstruction it has been assigned to several austenite grains. 

The code that I used is shown below:

              %% Correct Data
              rot = rotation.byEuler(90*degree,90*degree,0*degree);
              ebsd = rotate(ebsd,rot,'keepXY');
              
              % grain reconstruction
              [grains,ebsd.grainId] = calcGrains(ebsd('indexed'), 'angle', 3*degree);
              
              % remove small grains
              ebsd(grains(grains.grainSize < 4)) = [];
              
              % reidentify grains with small grains removed:
              [grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
              grains = smooth(grains,5);
              
              cs_bcc = ebsd('Iron bcc (old)').CS;
              cs_aus = ebsd.CSList{2};
              
              KS = orientation.map(Miller(1,1,1,cs_aus),Miller(0,1,1,cs_bcc),...
                    Miller(-1,0,1,cs_aus),Miller(-1,-1,1,cs_bcc));
              
              job = parentGrainReconstructor(ebsd,grains,KS) % define job
              
              job.p2c = orientation.KurdjumovSachs(job.csParent,job.csChild)
              
              job.calcParent2Child
              
              [fit,c2cPairs] = job.calcGBFit;
              
              [gB,pairId] = job.grains.boundary.selectByGrainId(c2cPairs);
              
              job.calcGraph('threshold',2.5*degree,'tolerance',2.5*degree);
              
              job.clusterGraph('inflationPower',1.6)
              
              job.calcParentFromGraph
              
              job.revert(job.grains.fit > 5*degree | job.grains.clusterSize < 15)
              
              for k = 1:3 % do this three times
              
                % compute votes
                job.calcGBVotes('noC2C');
              
                % compute parent orientations from votes
                job.calcParentFromVote('minFit',7.5*degree)
              end
              
              
              job.mergeSimilar('threshold',7.5*degree);
              
              job.mergeInclusions('maxSize',50);
              
              plot(job.parentGrains,job.parentGrains.meanOrientation)

I tried to modify the parameters in "job.mergeSimilar", "inflationPower", "job.calcGraph", and "job.revert", however, it does not seem to be successful. 

I would be very grateful if you can help me with this issue. The EBSD ".crc" file after compression is still too large (~30 MB) to be attached. I am happy to provide that by email just in case.

Thank you very much!

Best regards,
Hamid

**Comment by nyyssont** [comment]
Hello,

I agree with you, I think the reconstruction is faulty in this case.

It's quite possible that the clustering approach is not suitable for this particular case (or that particular grain boundary at the very least).

You could try the following:

```
job = parentGrainReconstructor(ebsd,grains) % define job
job.calcParent2Child; % get representative OR

job.calcGBVotes % compute votes from child grain boundaries
job.calcParentFromVote('minVotes',4)

for l = 1:2
job.calcGBVotes('noC2C') % compute votes ONLY from parent-child boundaries;
job.calcParentFromVote('minFit',5*degree) %compute parent orientations from votes
end
```

This method is based on finding initial prior austenite orientations based on the amount of neighbors that share a similar potential prior austenite orientation. These initial prior orientations are then used as "seeds" which are grown into full-size grains neighbor by neighbor. You can increase the amount of iterations in the for loop if necessary.

See:
Ranger, C., Tari, V., Farjami, S., Merwin, M.J., Germain, L. and Rollett, A. (2018) Austenite Reconstruction Elucidates Prior Grain Size Dependence of Toughness in a Low Alloy Steel. Metallurgical and Materials Transactions A 49: 4521-4535.

**Reply by nyyssont** [reply]
Another thought...

the division you see in the prior austenite grain could be due to annealing twin boundaries. You could check using the reconstructed grains:

```
% restrict to iron to iron phase transition
gB = grains_rec.boundary('Iron fcc','Iron fcc')

% select CSL(3) grain boundaries
gB3 = gB(angle(gB.misorientation,CSL(3,ebsd('Iron fcc').CS)) < 3*degree);

% overlay CSL(3) grain boundaries with grains plot
figure
plot(grains_rec)
hold on
plot(gB3,'lineColor','gold','linewidth',2,'DisplayName','CSL 3')
hold off
```

**Reply by HamidKamali** [reply]
Hi Tuomo,

Thank you very much for your quick response.

1. Tried the method to find initial prior austenite orientations based on the amount of neighbors with a similar potential prior austenite orientation. Still in that specific grain, the reconstruction identified several grains. I tried to modified parameters with a focus on the area of interest. The result of reconstruction on that specific grain is shown below: (The reconstruction result of the other grains could be faulty.)

![Mtex discussion1](https://user-images.githubusercontent.com/79619598/109237740-46fce000-7826-11eb-9bff-730cab810136.jpg)

2. plotted the grain boundaries colorized by the misorientation angle, as you can see below:

![Misorientation angle](https://user-images.githubusercontent.com/79619598/109236047-f59f2180-7822-11eb-910f-3da42b48c861.jpg)

It seems there is a twinning relationship in the grain.
Is there any possibility to verify the reconstruction result in that specific grain?

Thank you very much for your time.

Best regards,
Hamid

**Comment by HamidKamali** [comment, marked as answer]
Dear Mtex users,

Previously, I faced some challenges with the reconstruction of parent austenite grains in partially transformed microstructure to deformation-induced martensite of both types i.e. alpha prime and epsilon.

Recently, I tried ORTools (you can find it in **_https://github.com/frankNiessen/ORTools_**) which gives very high flexibility in the reconstruction of the parent phase (not only austenite) and got the fully reconstructed microstructure. Parent grain reconstruction, however, is only one out of many capabilities of it. There are numerous other interesting tools for microstructure analysis.

I wanted to share it with you and I hope you get the most out of it.

All the best,
Hamid
